### ISSUE TITLE
PasswordBoxes-Must-Have-Min-Length false positive on this regex

### ISSUE THREAD

**Issue by bmoore-msft** [Azure/arm-ttk#288]
The test doesn't detect this regex as having a minLength

```
              {
                "name": "adminPassword",
                "type": "Microsoft.Common.PasswordBox",
                "label": {
                  "password": "password",
                  "confirmPassword": "Confirm password"
                },
                "constraints": {
                  "required": true,
                  "regex": "^.{12,}",
                  "validationMessage": "Password must be at least 12 characters long"
                },
                "options": {
                  "hideConfirmation": false
                }
              }
```

**Comment by bmoore-msft** [comment]
@edburns - your PR will fix this case?

**Comment by edburns** [comment]
@bmoore-msft It will fix the case where the input regex to examine uses a positive lookahead.

> A positive lookahead, written as ‹(?=⋯)›, can be used at the beginning of the pattern to ensure that the string is within the target length range. The remainder of the regex can then validate the desired pattern without worrying about text length. 

I don't know what the poster means by "this regex" though.

**Comment by edburns** [comment]
Ah, if you mean ` "regex": "^.{12,}",` then, no. My patch does not address that particular case.  Personally, the way to check for "length" using regex is very open ended and application dependent. I view this test as a "best effort" test case. I assumed that was why the failure was classified as a "warning". @bmoore-msft .

**Comment by bmoore-msft** [comment]
yes, using a regex to validate a regex is an interesting exercise to say the least...

**Comment by edburns** [comment]
@bmoore-msft I do enjoy the programming equivalent of pointing your webcam at your screen though.

**Comment by StartAutomating** [comment]
This was fixed by #359
